GitUI/UserControls/RevisionGrid/Graph/RevisionGraph.cs

```diff
@@ -504,9 +504,20 @@ private RevisionGraphRevision[] BuildOrderedNodesCache(int currentRowIndex)
             _orderedUntilScore = int.MinValue;
             _reorder = false;
 
-            // Use a local variable, because the cached list can be reset
-            var localOrderedNodesCache = _nodes.ToArray();
-            Array.Sort(localOrderedNodesCache, (x, y) => x.Score.CompareTo(y.Score));
+            // Use a local variable, because the cached list can be reset.
+            // But the scores of the referenced nodes can still be rewritten as a last step.
+            // So if Sort complains, try again.
+            RevisionGraphRevision[] localOrderedNodesCache = _nodes.ToArray();
+            try
+            {
+                Array.Sort(localOrderedNodesCache, (x, y) => x.Score.CompareTo(y.Score));
+            }
+            catch (ArgumentException ex) when (ex.Message.Contains("IComparer.Compare()"))
+            {
+                localOrderedNodesCache = _nodes.ToArray();
```

Comment:
Is it worth to introduce a local function to remove the duplication? It took me some seconds to check if it was really a code duplication...

Reply:
I was aware of this and followed the [Rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)).
Though I have found a clearer (no duplication) and safer (cope with multiple rewrites, too) implementation.